Add documentation for new bosch alarm integration (#37458)
home-assistant/home-assistant.github.io: source/_integrations/bosch_alarm.markdown

---
title: Bosch Alarm
description: Integrate Bosch Alarms.
ha_category:
  - Alarm

> ha_release: 2025.2

Reviewer: ```suggestion
ha_release: 2025.3
```

ha_iot_class: Local Push
ha_config_flow: true
ha_codeowners:
  - '@mag1024'
  - '@sanjay900'
ha_domain: bosch_alarm
ha_platforms:
  - alarm_control_panel
ha_zeroconf: false
ha_integration_type: integration
---

The **Bosch Alarm Panel** {% term integration %} allows you to connect your [Bosch Alarm Panel](https://www.boschsecurity.com) to Home Assistant to control and monitor your Bosch Alarm Panel.

{% include integrations/config_flow.md %}

> Supported panels:

Reviewer: ```suggestion
## Supported devices
```

- _Solution 2000/3000/4000_
- B Series: _B3512/B4512/B5512/B6512_
- G Series: _B8512G/B9512G_
- _AMAX 2100/3000/4000_
- _D7412GV4/D9412GV4_ [^1]

[^1]: Firmware 2.0+

## Provided entities

The following {% term entities %} are provided:

- [AlarmControlPanel](#alarm-control-panel)

> ## Alarm Control Panel

Reviewer: ```suggestion
### Alarm Control Panel
```
this is probably part of the provided entities section?

This integration adds an Alarm Control Panel device for each configured area, with the ability to issue arm/disarm commands.
This entity reports state (_disarmed_, _armed_away_, etc.), and contains custom attributes _ready_to_arm_ (_no_|_home_|_away_), and a _faulted_points_ counter.

## Authentication

The primary means of authentication for the _Mode 2_ API is the _Automation_ passcode. It needs to be at least 10 characters long, and it is different from the _User_ code -- a shorter numeric pin used to arm/disarm the panel.
The integration will prompt for the required passcodes, which depend on the panel type.

| Panel | Code |
| --- | --- |
| Solution | User [^2] |
| B Series | Automation |
| G Series | Automation |
| AMAX | Both |

[^2]: The user needs to have the "master code functions" authority if you wish to interact with history events.

> ⚠️ Since the _Mode 2_ automation user has "superuser" privileges, it bypasses the regularly configured alarm pin: you will _not_ be prompted for a _User_ code when arming/disaming through the integration.

Reviewer: ```suggestion
{% important %}
Since the _Mode 2_ automation user has "superuser" privileges, it bypasses the regularly configured alarm pin: you will _not_ be prompted for a _User_ code when arming/disarming through the integration.
{% endimportant %}
```

---
title: Bosch Alarm
description: Integrate Bosch Alarms.
ha_category:
  - Alarm

> ha_release: 2025.3

Reviewer 2: ```suggestion
ha_release: 2025.4
```

ha_iot_class: Local Push
ha_config_flow: true
ha_codeowners:
  - '@mag1024'
  - '@sanjay900'
ha_domain: bosch_alarm
ha_platforms:
  - alarm_control_panel

> ha_zeroconf: false

Reviewer 2: ```suggestion
```

ha_integration_type: integration
---

The **Bosch Alarm Panel** {% term integration %} allows you to connect your [Bosch Alarm Panel](https://www.boschsecurity.com) to Home Assistant to control and monitor your Bosch Alarm Panel.

{% include integrations/config_flow.md %}

## Supported devices

- _Solution 2000/3000/4000_
- B Series: _B3512/B4512/B5512/B6512_
- G Series: _B8512G/B9512G_
- _AMAX 2100/3000/4000_
- _D7412GV4/D9412GV4_ [^1]

[^1]: Firmware 2.0+

## Provided entities

The following {% term entities %} are provided:

> - [AlarmControlPanel](#alarm-control-panel)

Reviewer 2: It's not a camel case thing

### Alarm Control Panel

This integration adds an Alarm Control Panel device for each configured area, with the ability to issue arm/disarm commands.

> This entity reports state (_disarmed_, _armed_away_, etc.), and contains custom attributes _ready_to_arm_ (_no_|_home_|_away_), and a _faulted_points_ counter.

Reviewer 2: no counter?

Author: Ah yep, removed that as i was told to expose those as sensors instead of attributes, so ill add them in a different pull later.
